src/typing/fields.ml

```diff
@@ -350,7 +350,7 @@ let rec type_field cfg ctx e i p mode =
 				(* the abstract field is not part of the field list, which is only true when it has no expression (issue #2344) *)
 				display_error ctx ("Field " ^ i ^ " cannot be called directly because it has no expression") p;
 			| _ ->
-				display_error ctx (StringError.string_error i (string_source t) (s_type (print_context()) t ^ " has no field " ^ i)) p;
+				display_error ctx (StringError.string_error i (string_source t) (s_type (print_context()) t ^ " has no field " ^ i)) e.epos;
```

Comment:
I think correct position should be calculated as a diff between `p` (which encloses whole expression) and `e.epos` (which encloses the part of expression without the field itself). This way reported position would point to the field name only.

Reply:
Yeah, did it earlier (in ~MORDOR~ `handle_efield`'s `loop`)